### DISCUSSION TITLE
Minimize delay from alsa inpu plugin

### DISCUSSION THREAD

**Discussion by aratar-orome** [MusicPlayerDaemon/MPD#2072] [Q&A]
Hi, 

I playback music from my line in via mpd. For that I have added an alsa input stream with

`mpc alsa://hw3,0`

Unfortunatly, I have a playback delay of ca. 3 seconds.

When I pipe the line in directly to the sound card I have no markable delay:

`arecord -Dhw:3 -f cd | aplay -Dstereo1`

Can you give me an advice to reduce the delay?

Thanks in advance,

Andreas

**Comment by MaxKellermann** [comment]
MPD is optimized for reliability, not for low latency. It's a trade-off, you can't have both, and for a music player, reliability is most important.

**Reply by aratar-orome** [reply]
Thank you very much Max for the fast response. Now I know that I do not need to try out other solutions. I am also happy with the current one.
